projects/exui-common-lib/src/lib/components/generic-filter/generic-filter.component.html

```diff
@@ -1,40 +1,62 @@
-<div class="govuk-form-group">
-  <div class="xui-generic-filter" *ngFor="let field of config.fields">
-    <div class="govuk-form-group" [ngClass]="{'form-group-error': field.displayMinSelectedError || field.displayMaxSelectedError}">
-      <h3 *ngIf="field.title" class="govuk-heading-s">{{field.title}}</h3>
-      <p class="govuk-body" *ngIf="field.subTitle">{{field.subTitle}}</p>
-      <div class="govuk-body" [ngSwitch]="field.type">
-        <ng-container *ngSwitchCase="'checkbox'">
-          <div class="govuk-checkboxes govuk-checkboxes--small">
-            <div *ngFor="let item of field.options" class="govuk-checkboxes__item">
-              <input type="checkbox" [name]="'checkbox_' + item.key" [id]="'checkbox_' + item.key" [value]="item.key" [checked]="isSelected(field.name, item.key)" (change)="onSelectionChange($event)" class="govuk-checkboxes__input" [attr.field]="field.name"/>
-              <label [for]="'checkbox_' + item.key" class="govuk-label govuk-checkboxes__label" [attr.field]="field.name">{{item.label}}</label>
+<form [formGroup]="form" (ngSubmit)="applyFilter(form)">
+    <div *ngFor="let field of config.fields">
+      <div class="govuk-form-group xui-generic-filter"
+           [hidden]="hidden(field, form)"
+           [ngClass]="{'form-group-error': submitted && (form.get(field.name).errors?.minLength || form.get(field.name).errors?.maxLength)}">
+        <h3 *ngIf="field.title" class="govuk-heading-s">{{field.title}}</h3>
+        <p class="govuk-body" *ngIf="field.subTitle">{{field.subTitle}}</p>
+        <div class="govuk-body" [ngSwitch]="field.type">
+          <ng-container *ngSwitchCase="'select'">
+            <select class="govuk-select" [name]="'select_' + field.name" [id]="'select_' + field.name" [formControlName]="field.name">
+              <ng-container *ngFor="let item of field.options">
+                <option class="govuk-radios__item" [value]="item.key">{{item.label}}</option>
+              </ng-container>
+            </select>
+          </ng-container>
+          <ng-container *ngSwitchCase="'checkbox'">
+            <div class="govuk-checkboxes govuk-checkboxes--small" [formGroupName]="field.name">
+              <div *ngFor="let item of field.options; let i = index" class="govuk-checkboxes__item">
+                <input type="checkbox" class="govuk-checkboxes__input"
+                       [formControlName]="i"
```

Comment:
could this formControlName be something descriptive rather than just the index? like checkbox-i.

Reply:
but you still can use the checkbox with index as a template in the ts like `checkbox-${i}` as the formControlName to keep the checkbox state, right?

Reply:
Couldn't get it to work 😩. Can you approve as per our discussion.